arwen/host/arwen.go

```diff
@@ -415,6 +421,11 @@ func (host *vmHost) RunSmartContractCall(input *vmcommon.ContractCallInput) (vmO
 				log.Error("VM execution panicked", "error", r, "stack", "\n"+string(debug.Stack()))
 				err = arwen.ErrExecutionPanicked
 				host.Runtime().CleanInstance()
+			} else {
+				err = host.Runtime().EndExecution()
+				if err != nil {
+					vmOutput = nil
```

Comment:
The error returned by `host.Runtime().EndExecution()` comes very late and it can happen even if the VM already produced a correct `VMOutput`.

Note that `host.Runtime().EndExecution()` _**never returns an error**_ in production: it is prevented by a global constant set to `false` in `runtime.go`. It is only used as a memory leak detector, which is disabled.